platform/security/platform-security-core-api/src/main/java/ddf/security/samlp/SamlProtocol.java

```diff
@@ -478,7 +478,7 @@ public static EntityDescriptor createSpMetadata(
       AssertionConsumerService assertionConsumerServicePaos =
           assertionConsumerServiceBuilder.buildObject();
       assertionConsumerServicePaos.setBinding(PAOS_BINDING);
-      assertionConsumerServicePaos.setIndex(acsIndex++);
```

Comment:
Why was this changed, did we never use the updated value?

Reply:
It's redefined every time this method is called so this last increment didn't really do anything.

Reply:
What @peterhuffer said

Reply:
The only issue I see is if someone decides that add another if statement after this one similar to the previous ones. The index at this point would be invalid as he might forget to go back and re-add the ++

Reply:
Yeah I would say it seems odd to *not* do it on the very last statement.  Feels like we should leave it in as a convenience to the next developer that comes along, or refactor it such that we are not doing it this way at all.  ( like by creating an IntSupplier and using the supplier to generate numbers for each index (`IntStream.iterate(0, i -> i + 1);`)